src/api/commands/bid/BidSearchCommand.ts

```diff
@@ -36,31 +38,59 @@ export class BidSearchCommand extends BaseCommand implements RpcCommandInterface
      */
     @validate()
     public async execute( @request(RpcRequest) data: RpcRequest): Promise<Bookshelf.Collection<Bid>> {
+        let itemHash;
+        if (data.params.length < 1) {
+            itemHash = '*';
+        } else {
+            itemHash = data.params.shift();
+        }
+
+        let status;
+        if (data.params.length < 1) {
+            status = '*';
+        } else {
+            status = data.params.shift();
+        }
+
+        const searchArgs = {} as BidSearchParams;
+        if (itemHash !== '*') {
+            searchArgs.listingItemHash = itemHash;
+        }
+        if (status !== '*') {
+            searchArgs.action = status;
+        }
 
-        // if hash is specified
-        if (data.params[0]) {
-            const listingItem = await this.listingItemService.findOneByHash(data.params[0]);
 
-            return this.bidService.search({
-                listingItemId: listingItem.id,
-                action: data.params[1]
-            } as BidSearchParams);
+        if (data.params.length === 0) {
+            return await this.bidService.search(searchArgs);
         } else {
-            // searching all bids
-            return this.bidService.findAll();
+            let listOfBids: any = null;
+            for (const address of data.params) {
+                searchArgs.bidder = address;
+                const tmp = await this.bidService.search(searchArgs);
+                if (listOfBids !== null) {
```

Comment:
rather than looping through the addresses, modify BidSearchParams:
    public bidder: string;
->
    public bidders: string[];

then modify this.bidService.search to handle search for an array of bidder addresses.

in Bid.ts, modify this to loop through bidders and add OR
                if (options.bidder) {
                    qb.where('bids.bidder', '=', options.bidder);
                }